src/main/resources/db/changesets/20231129_snapshotrequests.yaml

```diff
@@ -0,0 +1,60 @@
+databaseChangeLog:
+  - changeSet:
+      id: snapshot_requests
+      author: rjohanek
+      changes:
+        - createTable:
+            tableName: snapshot_requests
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: dataset_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_dataset_snapshot_requests
+                    references: dataset(id)
+                    deleteCascade: true
+              - column:
+                  name: request_name
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: request_research_purpose
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: snapshot_builder_request
+                  type: jsonb
+                  constraints:
+                    nullable: false
+              - column:
+                  name: user_email
```

Comment:
I see this type of column is called `user_email` in `journal` but is called `created_by` in most other TDR tables. In this case maybe it should be `created_by` (or another name like `requester`). The requester is always the one creating the request record, for now, so `created_by` seems OK to me.